core/src/main/java/org/lflang/diagram/synthesis/LinguaFrancaSynthesis.java

```diff
@@ -1251,6 +1258,33 @@ private Collection<KNode> transformReactorNetwork(
       }
     }
 
+    // Connect watchdogs
+    Set<WatchdogInstance> watchdogs = new HashSet<>();
+    watchdogs.addAll(watchdogSources.keySet());
+    watchdogs.addAll(watchdogDestinations.keySet());
+
+    for (WatchdogInstance watchdog : watchdogs) {
+      KNode node = associateWith(_kNodeExtensions.createNode(), watchdog.getDefinition());
+      NamedInstanceUtil.linkInstance(node, watchdog);
+      _utilityExtensions.setID(node, watchdog.uniqueID());
+      nodes.add(node);
+      setLayoutOption(node, CoreOptions.PORT_CONSTRAINTS, PortConstraints.FIXED_SIDE);
+      Pair<KPort, KPort> ports = _linguaFrancaShapeExtensions.addWatchdogFigureAndPorts(node);
+      if (watchdog.getTimeout() != null) {
+        _kLabelExtensions.addOutsideBottomCenteredNodeLabel(
+            node, String.format("timeout: %s", watchdog.getTimeout().toString()), 7);
+      }
+      // connect source
+      for (KPort source : watchdogSources.get(watchdog)) {
+        connect(createDelayEdge(watchdog), source, ports.getKey());
+      }
+
+      // connect targets
+      for (KPort target : watchdogDestinations.get(watchdog)) {
+        connect(createDelayEdge(watchdog), ports.getValue(), target);
+      }
```

Comment:
You might want to enable layout annotation for watchdogs, as done [here](https://github.com/lf-lang/lingua-franca/blob/4bb000e4dc88988e40e681592a2fc16f89908a2b/core/src/main/java/org/lflang/diagram/synthesis/LinguaFrancaSynthesis.java#L1058) for timers.
The layout annotations should be applied after everything else that sets layout options finishes so after `addWatchdogFigureAndPorts` was called.

Adding a method to configure layout for watchdogs in LayoutPostProcessing, as done [here](https://github.com/lf-lang/lingua-franca/blob/4bb000e4dc88988e40e681592a2fc16f89908a2b/core/src/main/java/org/lflang/diagram/synthesis/LinguaFrancaSynthesis.java#L1057) might not be necessary and I will add this once a new ELK release may add more interesting ordering related layout options.

Reply:
Done in 521a50245